portal/server/helpers/map-facility-properties.ts

```diff
@@ -0,0 +1,38 @@
+import { FacilityDashboard, FACILITY_STAGE } from '@ukef/dtfs2-common';
+
+/**
+ * Maps the facility stage based on the current facility properties.
+ *
+ * @param {Facility} facility - The facility object to map the stage for.
+ * @returns {Facility} The facility object with the mapped stage.
+ *
+ * The function checks the `facilityStage` property of the input `facility`.
+ * If the `facilityStage` is `FACILITY_STAGE.RISK_EXPIRED`, it sets the `facilityStage`
+ * of the returned facility to `FACILITY_STAGE.RISK_EXPIRED`.
+ * Otherwise, it sets the `facilityStage` based on the `hasBeenIssued` property:
+ * - If `hasBeenIssued` is `true`, it sets the `facilityStage` to `FACILITY_STAGE.ISSUED`.
+ * - If `hasBeenIssued` is `false`, it sets the `facilityStage` to `FACILITY_STAGE.UNISSUED`.
+ */
+const mapFacilityStage = (facility: FacilityDashboard): FacilityDashboard => {
+  const mappedFacility = facility;
```

Comment:
Should clone this to avoid modifying `facility` object
```suggestion
  const mappedFacility = cloneDeep(facility);
```